pennylane/tape/qnode.py

```diff
@@ -683,3 +683,40 @@ def qfunc_decorator(func):
         return update_wrapper(qn, func)
 
     return qfunc_decorator
+
+
+def make_drawing(_qnode, charset="unicode"):
+    """Make a function that draws the given _qnode.
+
+    Example: Given this definition of a Qnode.
+
+    .. code-block:: python3
+
+        qml.enable_tape()
+
+        @qml.qnode(dev)
+        def circuit(a, w):
+            qml.Hadamard(0)
+            qml.CRX(a, wires=[0, 1])
+            qml.Rot(*w, wires=[1])
+            qml.CRX(-a, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+    We can draw the it like such:
+
+    >>> qml.make_drawing(circuit)(a=2.3, w=[1.2, 3.2, 0.7])
+    0: ──H──╭C────────────────────────────╭C─────────╭┤ ⟨Z ⊗ Z⟩
+    1: ─────╰RX(2.3)──Rot(1.2, 3.2, 0.7)──╰RX(-2.3)──╰┤ ⟨Z ⊗ Z⟩
+
+    Args:
+        _qnode: A ``pennylane.qnode``.
+    Returns:
+        A function that has the same arguement signature as ``qnode``. When called,
+        the function will draw the qnode.
+    """
+
+    def wrapper(*args, **kwargs):

```

Comment:
Should we do `@functools.wraps` here? I'm not 100% sure.

* If we don't, the user will simply see something obscure like `qnode.make_drawing.<lambda>.wrapper` on the terminal, with no documentation.

* If we do, the user will see their original QNode function docstring on the terminal.

It would be cool if we could wrap it in such a way as to preserve the original function signature/docstring, but modify it to include 'drawing only' in the description.

Reply:
I think wraps here is appropriate.